### ISSUE TITLE
Not all nodes removed from a group in Undo/Redo stack

### ISSUE THREAD

**Issue by andydandy74** [DynamoDS/Dynamo#8300]
## Dynamo version
1.3.1

## Operating system
Windows 7

## What did you do? 
Removed two nodes from this group:
![grafik](https://user-images.githubusercontent.com/3014437/32246477-4e4e0e1a-be7f-11e7-89ab-2a8b257ea06e.png)
Then hit Undo
Then moved the group

## What did you expect to see?
All nodes belonging to the group again

## What did you see instead?
![grafik](https://user-images.githubusercontent.com/3014437/32246520-7d859ae0-be7f-11e7-804a-a64cfa10ef4e.png)
Only one of the nodes re-joined the group. As far as I can tell from my testing, it is always the most recently created node.

**Comment by ramramps** [comment]
@andydandy74 can you try `undo` one more time and see whether the node is getting added to the group? 
I did the same as mentioned, but I had to do one `undo` extra to get the last node into the group.

**Comment by andydandy74** [comment]
@ramramps - yes, that does indeed work. It is higly impractical, though. What if I removed 20 nodes from the group? I would have to hit Redo 20 times...

**Comment by ramramps** [comment]
@andydandy74 no. the last before UNDO is for the group itself. Groups are created after the nodes. So, the last undo (which doesn't do anything) is for the group, but the node is not created yet. 
I will try with many nodes and see how it behaves.

**Comment by Dewb** [comment]
Described behavior is still reproducible in 2.x, though it looks like a display issue -- the node does rejoin the group, but the group's background is not redrawn to include the newly returned node.

**Comment by johnpierson** [comment]
Updating this (really old) ticket with findings from modern 4.x Dynamo.

Removing several nodes from a group at once requires several redos to re-add them to the group. It would be great if the undos were consolidated into 1.
